pages/firebase-firsttry.js

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from 'react';
+import { collection, onSnapshot } from 'firebase/firestore';
+import Link from 'next/link';
+
+import { db } from '../firebase/initFirebase';
+
+const FirebaseFirst = () => {
+  const [articleList, setArticleList] = useState([]);
+  const articlesCollectionRef = collection(db, 'articles');
+
+  useEffect(() => {
+    onSnapshot(articlesCollectionRef, (snapshot) => {
+      setArticleList(
+        snapshot.docs.map((doc) => ({
+          ...doc.data(),
+          id: doc.id,
+        })),
+      );
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Turn this rule off. I find the rule a bit weird overall.

Reply:
hmm, can I keep it? It makes me think about what I am doing :D